applications/PoromechanicsApplication/python_scripts/poromechanics_U_Pw_solver.py

```diff
@@ -129,21 +78,46 @@ def AddVariables(self):
             self.main_model_part.AddNodalSolutionStepVariable(KratosPoro.NODAL_JOINT_WIDTH)
             self.main_model_part.AddNodalSolutionStepVariable(KratosPoro.NODAL_JOINT_DAMAGE)
 
+        KratosMultiphysics.Logger.PrintInfo("UPwSolver", "Variables added correctly.")
 
-        print("Variables correctly added")
+    def ImportModelPart(self):
+        # we can use the default implementation in the base class
+        self._ImportModelPart(self.main_model_part,self.settings["model_import_settings"])
 
-    def GetMinimumBufferSize(self):
-        return 2
+    def PrepareModelPart(self):
 
-    def AddDofs(self):
+        # Set ProcessInfo variables
```

Comment:
isn't this done in the basclass of the AnalysisStage?

Reply:
They are initialized when first used ...
`TIME`: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/analysis_stage.py#L119-L121
`STEP`: In `AdvanceTimeStep` of the solver
`DELTA_TIME`: inside `ProcessInfo` when `CloneTImeStep` is called

Reply:
Thanks @philbucher !
In any case, I prefer to initialize those variables before filling the buffer and initializing the strategy...